api_app/analyzers_manager/models.py

```diff
@@ -0,0 +1,85 @@
+from django.db import models
+from django.contrib.postgres import fields as postgres_fields
+from cache_memoize import cache_memoize
+
+from intel_owl.secrets import get_secret
+from api_app.models import Job
+
+
+class Analyzer(models.Model):
+    CHOICES = (
+        ("file_analyzer", "file_analyzer"),
+        ("observable_analyzer", "observable_analyzer"),
+    )
+    analyzer_type = models.CharField(
+        max_length=50,
+        choices=CHOICES,
+    )
+    disabled = models.BooleanField(default=False, blank=False)
+    description = models.TextField()
+    python_module = models.CharField(max_length=128, blank=False, null=False)
+    config = postgres_fields.JSONField(default=dict, blank=False, null=False)
+
+    @property
+    def _cached_secrets(self) -> dict:
+        return self.__cached_secrets
+
+    @cache_memoize(100, args_rewrite=lambda o: o.pk)
+    def verify_secrets(self) -> dict:
+        verification = {
+            "configured:": False,
+            "error_message": None,
+            "missing_secrets": [],
+        }
+        for secret in self.secrets.all():
+            var = get_secret(secret.env_variable_key)
+            if var:
+                self.__cached_secrets[secret.name] = var
+            else:
+                verification["missing_secrets"].append(secret.name)
+
+        if len(verification["missing_secrets"]) == 0:
+            verification["configured"] = True
+        else:
+            total_secrets = self.secrets.all().count()
+            verified_secrets = total_secrets - len(verification["missing_secrets"])
+            verification[
+                "error_message"
+            ] = f'{verification["missing_secrets"]} missing ({verified_secrets} of {total_secrets} satisfied)'  # noqa: E501
+
+        return verification
+
+
+class AnalyzerReport(models.Model):
+    analyzer = models.ForeignKey(
+        Analyzer, related_name="reports", on_delete=models.CASCADE
+    )
+    job = models.ForeignKey(
+        Job, related_name="analyzer_reports", on_delete=models.CASCADE
+    )
+
+    success = models.BooleanField(default=False)
```

Comment:
Instead of a boolean field like this, i think better would be to have a `status = models.CharField(choices=...)`, with choices like: `pending`, `running`, `failed`, `success`.